### PR TITLE
feat: add GN (Generate Ninja)

### PR DESCRIPTION
**Pull request by amaanq**
https://gn.googlesource.com/gn/+/main/docs/reference.md#grammar

### CODE REVIEW

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -69,7 +69,7 @@ end
 ---@field experimental boolean|nil
 ---@field readme_name string|nil
 
----@type ParserInfo[]
```
Comment (lucario387) [inline lua/nvim-treesitter/parsers.lua:72]: shouldn't this still be type?

Comment (amaanq) [reply lua/nvim-treesitter/parsers.lua:72]: oops, that's extraneous but I did that to shut lua_ls up :sweat_smile:

File: queries/gn/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+(scope_access field: (identifier) @field)
+
+; Literals
+
+(string) @string
+
+(escape_sequence) @string.escape
+
+(expansion) @none
```
Comment (lucario387) [inline queries/gn/highlights.scm:53]: Is `@none` really needed here considering it's just to nullify highlights, which can be done by not capturing at all? 🤔

Comment (amaanq) [reply queries/gn/highlights.scm:53]: expansions are potentially inside strings, it's to remove the string hl

**Review by lucario387 — APPROVED** [review]

**Pull request merged**